addons/web/static/src/core/debug/profiling/profiling_service.js

```diff
@@ -46,9 +46,13 @@ const profilingService = {
                 params
             );
             const resp = await orm.call("ir.profile", "set_profiling", [], kwargs);
-            state.session = resp.session;
-            state.collectors = resp.collectors;
-            state.params = resp.params;
+            if (resp.type == "ir.actions.act_window") {
+                env.services.action.doAction(resp);
```

Comment:
Shouldn't this return immediately? Since it doesn't wait for the action resolve I'm not sure triggering the UPDATE makes much sense.

Reply:
I thought that the form will be updated a first time by the click (checking the checkbox), and the update will "reset" it to the previous state (uncheck the checkbox since the operation failed) but this is not really usefull since the debug menu is closed.

Reply:
Wait, is the method `doAction` synchronuous?  In doubt, you better `await` on its result, or return it.

Reply:
I don't get why we would need to await or return here. Executing the action can be done "after" and don't need to be synchronous I think.

Reply:
@Xavier-Do I suggested returning because in the original code the `update` and `updateDebugIcon` code where outside the fallback and I assumed this would either not do anything useful or misbehave.